### PR TITLE
I decimated the dotnet backend in favour of javascript because that f…

### PR DESCRIPTION
**Pull request by ryancosans**
…elt better (even though my javascript is terrible)

refactors obviously required for the js :thumbsup:

### CODE REVIEW

File: wwwroot/index.html [hunk 1]
```diff
@@ -185,26 +186,54 @@
...
 
                     document.getElementById("json").appendChild(feature);
-                    localStorage.setItem(item.name, item.enabled);
+                    localStorage.setItem(item.name, JSON.stringify(item));
                 });
             }
         };
 
-        xhr.open("GET", "http://localhost:5001/getFeatures", true);
+        xhr.open("GET", "http://localhost:5005/features", true);
```
Comment (masduo) [inline wwwroot/index.html]: Assume you're planning to swap these resource endpoints with the setup.json values

File: wwwroot/index.html [hunk 2]
```diff
@@ -185,26 +186,54 @@
...
+        subXhr.open("GET", "http://localhost:5005/subscribers", true);
+        subXhr.send();
+
         function updateFeatures(name) {
-            var featureStatus = localStorage.getItem(name);
+            var feature = JSON.parse(localStorage.getItem(name));
+            feature.enabled = !feature.enabled; //flip the switch
             var xhr = new XMLHttpRequest();
             xhr.onreadystatechange = function () {
                 if (xhr.readyState == DONE && xhr.status == OK) {
```
Comment (masduo) [inline wwwroot/index.html]: Would be good to put a note on screen for unhappy flows, connection failed, json was malformed etc.

File: wwwroot/index.html [hunk 3]
```diff
@@ -185,26 +186,54 @@
...
-                    console.log(featureStatus != "true");
+
+                    localStorage.setItem(name, JSON.stringify(feature));
+                    var subscribers = JSON.parse(localStorage.getItem("subscribers"));
+                    subscribers.map(function (subscriber) {
+                        var xhr = new XMLHttpRequest();
+
+                        xhr.onreadystatechange = function () {
+                            if (subXhr.readyState == DONE && subXhr.status == OK) {
+                                console.log("yay!");
```
Comment (masduo) [inline wwwroot/index.html]: Shouldn't be subXhr.readyState,
Also be a bit more descriptive, like "hell yeah, subscriber x is in the know...!"

Comment (masduo) [inline wwwroot/index.html]: Which also means the subscribers' post response will have to have a body with info about the subscriber.
Also the failure of this call is important, need to have a plan for unhappy flow here.

File: wwwroot/index.html (hunk 1 above)
Comment (ryancosans) [inline wwwroot/index.html]: absolutely I will work on that next :wink:

**Pull request merged**